lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -3693,7 +3693,10 @@ static bool ReplaceUseOfZeroInit(Instruction *I, Value *V, DominatorTree &DT,
       if (ReplaceUseOfZeroInit(I, UI, DT, Reachable))
         continue;
     } else if (LoadInst *LI = dyn_cast<LoadInst>(UI)) {
-      LI->replaceAllUsesWith(ConstantAggregateZero::get(LI->getType()));
+      // Replace uses of the load with a constant zero.
+      Constant *replacement = Constant::getNullValue(LI->getType());
+      DXASSERT(replacement != nullptr, "missing case when replacing with zero");
```

Comment:
nit: [LLVM already asserts for invalid types](https://llvm.org/doxygen/Constants_8cpp_source.html#l00370), so this assert is unnecessary.

Reply:
LGTM - thanks